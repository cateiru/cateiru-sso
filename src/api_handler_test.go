package src_test

import (
	"net/http"
	"testing"

	"github.com/cateiru/cateiru-sso/src"
	"github.com/cateiru/go-http-easy-test/v2/easy"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestApiOpenidConfigurationHandler(t *testing.T) {
	h := NewTestHandler(t)

	t.Run("取得可能", func(t *testing.T) {
		m, err := easy.NewMock("/", http.MethodGet, "")
		require.NoError(t, err)

		c := m.Echo()

		err = h.ApiOpenidConfigurationHandler(c)
		require.NoError(t, err)

		response := src.OpenidConfiguration{}
		require.NoError(t, m.Json(&response))

		snaps.MatchSnapshot(t, response)
	})
}

func TestJwksJsonHandler(t *testing.T) {
	h := NewTestHandler(t)

	t.Run("取得可能", func(t *testing.T) {
		m, err := easy.NewMock("/", http.MethodGet, "")
		require.NoError(t, err)

		c := m.Echo()

		err = h.JwksJsonHandler(c)
		require.NoError(t, err)

		data := m.W.Body.Bytes()

		snaps.MatchSnapshot(t, string(data))
	})
}
