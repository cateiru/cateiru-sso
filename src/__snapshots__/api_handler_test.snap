
[TestApiOpenidConfigurationHandler/取得可能 - 1]
src.OpenidConfiguration{
    Issuer:                                     "http://cateiru.test",
    AuthorizationEndpoint:                      "http://cateiru.test/oidc",
    TokenEndpoint:                              "http://localhost:8080/v2/token",
    UserinfoEndpoint:                           "http://localhost:8080/v2/userinfo",
    JwksUri:                                    "http://cateiru.test/.well-known/jwks.json",
    RegistrationEndpoint:                       "http://localhost:8080/v2/register",
    ScopesSupported:                            {"openid", "profile", "email"},
    ResponseTypesSupported:                     {"code"},
    ResponseModesSupported:                     nil,
    GrantTypesSupported:                        nil,
    AcrValuesSupported:                         nil,
    SubjectTypesSupported:                      {"public", "pairwise"},
    IdTokenSigningAlgValuesSupported:           {"RS256"},
    IdTokenEncryptionAlgValuesSupported:        nil,
    IdTokenEncryptionEncValuesSupported:        nil,
    UserinfoSigningAlgValuesSupported:          nil,
    UserinfoEncryptionAlgValuesSupported:       nil,
    UserinfoEncryptionEncValuesSupported:       nil,
    RequestObjectSigningAlgValuesSupported:     nil,
    RequestObjectEncryptionAlgValuesSupported:  nil,
    RequestObjectEncryptionEncValuesSupported:  nil,
    TokenEndpointAuthMethodsSupported:          {"client_secret_basic"},
    TokenEndpointAuthSigningAlgValuesSupported: nil,
    DisplayValuesSupported:                     nil,
    ClaimTypesSupported:                        nil,
    ClaimsSupported:                            {"iss", "sub", "aud", "exp", "iat", "auth_time", "nonce", "acr", "azp", "name", "given_name", "family_name", "middle_name", "nickname", "preferred_username", "picture", "email", "email_verified", "gender", "birthdate", "zoneinfo", "locale", "updated_at"},
    ServiceDocumentation:                       "https://github.com/cateiru/cateiru-sso",
    ClaimsLocalesSupported:                     {"ja-JP"},
    UiLocalesSupported:                         {"ja-JP"},
    ClaimsParameterSupported:                   false,
    RequestParameterSupported:                  false,
    RequestUriParameterSupported:               false,
    RequireRequestUriRegistration:              false,
    OpPolicyUri:                                "http://cateiru.test/policy",
    OpTosUri:                                   "http://cateiru.test/policy",
}
---
